### ISSUE TITLE
-dtd option

### ISSUE THREAD

**Issue by y2kenny** [highsource/jaxb-tools#247]
When I try to parse a dtd (using sourceType dtd), I get the following error

```
org.xml.sax.SAXParseException: Are you trying to compile DTD? Support for DTD is experimental. You may enable it by using the -dtd option.
```

How do I pass the -dtd option to xjc via this plugin?

**Comment by stephan-vollmer** [comment]
I'm having the same issue (plugin version 0.15.1 with Java 11.0.16).

I tried to specify the `-dtd` argument, but the warning still appears:
```
<configuration>
    <generatePackage>...</generatePackage>
    <generateDirectory>${project.build.directory}/generated-sources/xjc1</generateDirectory>
    <schemaIncludes>
        <schemaInclude>filename.dtd</schemaInclude>
    </schemaIncludes>
    <markGenerated>true</markGenerated>
    <args>
        <arg>-dtd</arg>
    </args>
</configuration>
```

**Comment by gigermocas** [comment]
you have a _schemaLanguage_ configuration option:
```
<schemaLanguage>dtd</schemaLanguage>
```

**Comment by y2kenny** [comment]
> you have a _schemaLanguage_ configuration option:
> 
> ```
> <schemaLanguage>dtd</schemaLanguage>
> ```

I have that but it did not remove the error.

**Comment by dkowis** [comment]
I have both <schemaLanguage>dtd</schemaLanguage> and tried the args bit, unfortunately it's not working.

**Comment by mattrpav** [comment]
There is a newer dtd dependency out there. I'm cutting a 0.15.2 first to try to clear up more critical issues than this WARN msg

**Comment by laurentschoelens** [comment]
@y2kenny / @stephan-vollmer : could you please provide an example file to test your issue ? Thanks

**Comment by y2kenny** [comment]
> @y2kenny / @stephan-vollmer : could you please provide an example file to test your issue ? Thanks

This is the dtd file I have used before:
https://gerrit.googlesource.com/plugins/manifest-subscription/+/refs/heads/master/src/main/resources/schemas/manifest.dtd

**Comment by laurentschoelens** [comment]
@y2kenny : found the culprit
A change has been made (maybe a typo in earlier version of xjc), that made schemaLanguage != XMLSCHEMA complains about DTD / WSDL generating code since it's "experimental".
See : https://github.com/eclipse-ee4j/jaxb-ri/commit/94a5a1f72caa81136e9ff2a91623d19c4748aab8

There's nothing (in my opinion) we can do to get rid of this message (it's just a warning for 'experimental' features that still exists in v4 of xjc, with more generation warning about rng / rnc files) and your classes are just generated fine (tested against current 2.0.3-SNAPSHOT).
@mattrpav : do you agree with me ?

**Comment by laurentschoelens** [comment]
Found this issue in jaxb-ri impl that was treated in main : https://github.com/eclipse-ee4j/jaxb-ri/issues/1701
Asked if any backport for 2.3.x branch will be done.

**Comment by laurentschoelens** [comment]
@y2kenny : this should be fixed in 4.0.4 based-jaxb-tools and 2.3.9 based jaxb-tools since Lukas is doing some backports in EE8 branch of JAXB-RI
We should be able to test and validate it with jaxb-tools 2.X branch and then release a fixed version based on latest jaxb-ri

**Comment by laurentschoelens** [comment]
@y2kenny / @stephan-vollmer / @dkowis / @gigermocas : sorry for the spam 😄 This was fixed in v4.0.2 of jaxb-tools

This will also be fixed in v2.0.10

**Comment by laurentschoelens** [comment]
Will be closed in next v2.x release. Fixed by #437
